src/ints/mouse_dos_driver.cpp

```diff
@@ -0,0 +1,1440 @@
+{
+    if (bit::is(x, b15))
+        // 0xffff for -1, 0xfffe for -2, etc.
+        return static_cast<int16_t>(x - 0x10000);
+    else
+        return static_cast<int16_t>(x);
+}
+
+static uint16_t GetPosX()
+{
+    return static_cast<uint16_t>(std::lround(pos_x)) & state.gran_x;
+}
+
+static uint16_t GetPosY()
+{
+    return static_cast<uint16_t>(std::lround(pos_y)) & state.gran_y;
+}
+
+// ***************************************************************************
+// Data - default cursor/mask
+// ***************************************************************************
+
+static constexpr uint16_t DEFAULT_TEXT_AND_MASK = 0x77FF;
+static constexpr uint16_t DEFAULT_TEXT_XOR_MASK = 0x7700;
+
+static uint16_t DEFAULT_SCREEN_MASK[CURSOR_SIZE_Y] = {
+    0x3FFF, 0x1FFF, 0x0FFF, 0x07FF, 0x03FF, 0x01FF, 0x00FF, 0x007F,
+    0x003F, 0x001F, 0x01FF, 0x00FF, 0x30FF, 0xF87F, 0xF87F, 0xFCFF
+};
+
+static uint16_t DEFAULT_CURSOR_MASK[CURSOR_SIZE_Y] = {
+    0x0000, 0x4000, 0x6000, 0x7000, 0x7800, 0x7C00, 0x7E00, 0x7F00,
+    0x7F80, 0x7C00, 0x6C00, 0x4600, 0x0600, 0x0300, 0x0300, 0x0000
+};
+
+// ***************************************************************************
+// Text mode cursor
+// ***************************************************************************
+
+// Write and read directly to the screen. Do no use int_setcursorpos (LOTUS123)
+extern void WriteChar(uint16_t col, uint16_t row, uint8_t page, uint8_t chr,
+                      uint8_t attr, bool useattr);
+extern void ReadCharAttr(uint16_t col, uint16_t row, uint8_t page, uint16_t *result);
+
+void RestoreCursorBackgroundText()
+{
+    if (state.hidden || state.inhibit_draw)
+        return;
+
+    if (state.background) {
+        WriteChar(state.backposx,
+                  state.backposy,
+                  real_readb(BIOSMEM_SEG, BIOSMEM_CURRENT_PAGE),
+                  state.backData[0],
+                  state.backData[1],
+                  true);
+        state.background = false;
+    }
+}
+
+void DrawCursorText()
+{
+    // Restore Background
+    RestoreCursorBackgroundText();
+
+    // Check if cursor in update region
+    auto x = GetPosX();
+    auto y = GetPosY();
+    if ((y <= state.update_region_y[1]) && (y >= state.update_region_y[0]) &&
+        (x <= state.update_region_x[1]) && (x >= state.update_region_x[0])) {
+        return;
+    }
+
+    // Save Background
+    state.backposx = static_cast<uint16_t>(x >> 3);
+    state.backposy = static_cast<uint16_t>(y >> 3);
+    if (state.mode < 2)
+        state.backposx = static_cast<uint16_t>(state.backposx >> 1);
+
+    // use current page (CV program)
+    uint8_t page = real_readb(BIOSMEM_SEG, BIOSMEM_CURRENT_PAGE);
+
+    if (state.cursor_type == MouseCursor::Software) {
+        uint16_t result;
+        ReadCharAttr(state.backposx, state.backposy, page, &result);
+        state.backData[0] = (uint8_t)(result & 0xff);
+        state.backData[1] = (uint8_t)(result >> 8);
```

Comment:
Would this benefit from bit_view? I think that's a good use-case for the bit-shift operator (here you want to get the high-byte, as opposed to performing a division).

Reply:
OK, used the bit_view.